### ISSUE TITLE
Natural Earth projection troubles

### ISSUE THREAD

**Issue by IanMayo** [debrief/debrief#1080]
Here is the view of the world that my analysts are used to:
![mercator](https://cloud.githubusercontent.com/assets/1108513/5957980/d7caa39c-a7b7-11e4-91ba-125a77336a8e.png)

As you can see, each grid square is about twice as tall as it is wide.  This reflects the "real world" proportion of the lines if they were physically painted onto the earth.  The data is being "projected" so that it's approximates to the proportions of the real (curved) world.

Here is the view in our Natural Earth branch:
![flatprojection](https://cloud.githubusercontent.com/assets/1108513/5957987/e81f072e-a7b7-11e4-89aa-02c57c15d28c.png)

The grid squares are square.  This means the height (latitude) of the land is being squashed to 50% of its real size.  It make the analysts data look wrong.  A square are in real life will look really flat and squashed in this view.

We need a strategy to overcome this.   I can think of the following:
1. find a mercator projection that does work with NE data
2. edit the shape files in [QGis] (http://gis.stackexchange.com/questions/63444/reprojecting-natural-earth-admin-0-countries-shapefile-to-web-mercator) so that there isn't any data near the Poles, and still use 4326.  I'm happy to ask an oDesk uDig expert to do this.
3. accept that the 200m layer won't appear, but give the GeoTools image a light-blue backdrop.  This will show through where the 200m layer should be, and we can still use 4326.  This will give us a blue ocean at 10M & 50M resolutions too.
4. anything else?

**Comment by IanMayo** [comment]
I've done some minor refactoring to clarify what we're using the projections for in the sources:  https://github.com/pecko/debrief/commit/17b5a2937aa1425fb0ca4f2f95413e038795b09b

**Comment by pecko** [comment]
@IanMayo Could you, please, explain how you get these screenshots (especially first)?

**Comment by IanMayo** [comment]
Sure.  I've refactored GtProjection in your branch, so that we have more control over the transforms.  The first image is using this projection:
![projectionvalues](https://cloud.githubusercontent.com/assets/1108513/5962500/53817910-a7dc-11e4-98c1-14d91a1d039e.png)

**Comment by IanMayo** [comment]
@pecko - I've found an oDesk QGis user who is going to trim the data for us.  He's going to start with the 10M_Admin_Countries file.  Hopefully, if he "trims" Antarctica so that it only goes down to 89 Degs South it will work for us.

If it does, he can apply the same edit to the other Shapefiles.

But, you are quite welcome to consider other strategies.  I think Option 3 (blue background) *may* be an easy fix.

**Comment by pecko** [comment]
I'm not sure that I understand option 3, but will check this issue.

**Comment by IanMayo** [comment]
Option 3.  If you compare the two screenshots in the first comment, you'll see that the 0m bathy shapefile isn't rendered - we just get a white background.

I think it's the 0m that isn't rendered because it's the 0m layer that extends to the North Pole.

But, if we paint the GeoTools background in that shade before we start rendering the shapefiles on top of it, then we will effectively get the correct image (since the 0m blue shade will show through).

That's my idea, anyway.

**Comment by pecko** [comment]
You mean something like the following (World projection: 3395, geotools background: Color(143, 216, 216)):

![issue1080](https://cloud.githubusercontent.com/assets/534584/5965403/29f61e78-a7fc-11e4-82ea-0f678d526d3d.png)

**Comment by pecko** [comment]
I have created the commit: https://github.com/pecko/debrief/commit/82de7a1db28965e2e5d02bf5248c8a65ff6ed82a in order to you review.

BTW Seems zooming is fine too.

**Comment by IanMayo** [comment]
Yes, but the color to use is the color from the 0m bathy SLD (the 0 color in your key).

**Comment by pecko** [comment]
Fixed by https://github.com/pecko/debrief/commit/9197b7b7a10c985a316b9625d3b100ab4e14cffa

Screencast:

![issue1080a](https://cloud.githubusercontent.com/assets/534584/5966584/f81357fe-a805-11e4-9dc8-067501bbf437.png)

**Comment by IanMayo** [comment]
Ok, we've got a valid solution for Option 3.

As we zoom out, and use 50M and 110M, do we still see an Ocean?

**Comment by pecko** [comment]
> As we zoom out, and use 50M and 110M, do we still see an Ocean?

I have added this background as geotools layer. It is always shown behind all other geotools layers.
It mean, yes - it is still see as bathy 0 color when some layer has background (Ocean).

Of course, we can change this behaviour. It can test scale/resolution and be shown only in some resolution(s).

**Comment by pecko** [comment]
Screencast 2:

![issue1080b](https://cloud.githubusercontent.com/assets/534584/5966691/e75a8f62-a806-11e4-9128-da6778ae8ed1.png)

**Comment by IanMayo** [comment]
I look forward to having a play tomorrow.

**Comment by pecko** [comment]
OK. Please review zooming too.

I have also rebased the issue984_naturalearth branch again (it is obliged because the master branch is changed), so travis build works: https://travis-ci.org/debrief/debrief/builds/48817144

**Comment by IanMayo** [comment]
@pecko - thanks for the background fill fix. You've provided a good implementation.  I suspect it should have gone into its own branch, since it's an experiment.

I suspect that trimming the Shapefile will still provide a tidier solution:  our *Solution 3* strategy means the still don't see the Antarctica land mass.

If *Solution 2* does work, then that will give us a much tidier implementation. I hope to find out about Solution 2 in the next couple of hours.

**Comment by IanMayo** [comment]
@pecko  - the updated Shapefiles have started coming through, and so far they work perfectly with EPSG:4326.   I'll keep you updated.

**Comment by pecko** [comment]
@IanMayo You mean they work when using WORLD_PROJECTION = EPSG:4326?
I suppose that you would upload these files to the https://github.com/debrief/ref_data repository.

**Comment by IanMayo** [comment]
Yes, exactly.  Yes, I'm changing the files in that repo as I check them.  I haven't done a Push yet.  Will do shortly.

**Comment by pecko** [comment]
BTW Does zooming work?

**Comment by IanMayo** [comment]
Zooming is better, but not perfect. I think I'll put it into another issue.

**Comment by pecko** [comment]
new screencast (grid are square):

![issue1080c](https://cloud.githubusercontent.com/assets/534584/5975821/56a50882-a886-11e4-8744-9ecf01ed736d.png)

**Comment by IanMayo** [comment]
I think we're done with this now.
